src/LockTableTest.cc

```diff
@@ -0,0 +1,155 @@
+            lockTable.acquireLock(2 * i + 1);
+        }
+        for (uint64_t i = 0; i < LockTable::ENTRIES_PER_CACHE_LINE; i++) {
+            lockTable.releaseLock(2 * i + 1);
+        }
+        EXPECT_FALSE(lockTable.buckets[1].next == NULL);
+    }
+
+    ~LockTableTest()
+    {}
+
+    DISALLOW_COPY_AND_ASSIGN(LockTableTest);
+};
+
+TEST_F(LockTableTest, constructor) {
+    LockTable lt(16);
+    for (uint32_t i = 0; i < 16; i++) {
+        for (uint32_t j = 0; j < lt.ENTRIES_PER_CACHE_LINE; j++)
+            EXPECT_EQ(0UL, lt.buckets[i].entries[j]);
+        EXPECT_TRUE(lt.buckets[i].next == NULL);
+    }
+}
+
+TEST_F(LockTableTest, constructor_truncate) {
+    // This is effectively testing BitOps::powerOfTwoLessOrEqual.
+    EXPECT_EQ(0x0UL, LockTable(1).bucketIndexHashMask);
+    EXPECT_EQ(0x1UL, LockTable(2).bucketIndexHashMask);
+    EXPECT_EQ(0x1UL, LockTable(3).bucketIndexHashMask);
+    EXPECT_EQ(0x3UL, LockTable(4).bucketIndexHashMask);
+    EXPECT_EQ(0x3UL, LockTable(5).bucketIndexHashMask);
+    EXPECT_EQ(0x3UL, LockTable(6).bucketIndexHashMask);
+    EXPECT_EQ(0x3UL, LockTable(7).bucketIndexHashMask);
+    EXPECT_EQ(0x7UL, LockTable(8).bucketIndexHashMask);
+}
+
+TEST_F(LockTableTest, acquireLock) {
+    EXPECT_EQ(0UL, lockTable.buckets[1].entries[1]);
+    lockTable.acquireLock(1);
+    EXPECT_EQ(1UL, lockTable.buckets[1].entries[1]);
+}
+
+TEST_F(LockTableTest, isLockAcquired) {
+    EXPECT_EQ(0UL, lockTable.buckets[1].next->entries[1]);
+    EXPECT_FALSE(lockTable.isLockAcquired(1));
+    lockTable.buckets[1].next->entries[1] = 1;
+    EXPECT_EQ(1UL, lockTable.buckets[1].next->entries[1]);
+    EXPECT_TRUE(lockTable.isLockAcquired(1));
+}
+
+TEST_F(LockTableTest, releaseLock_basic) {
+    lockTable.buckets[0].entries[2] = 2;
+    lockTable.buckets[1].entries[1] = 1;
+
+    EXPECT_EQ(1UL, lockTable.buckets[1].entries[1]);
+    lockTable.releaseLock(1);
+    EXPECT_EQ(0UL, lockTable.buckets[1].entries[1]);
+
+    EXPECT_EQ(2UL, lockTable.buckets[0].entries[2]);
+    lockTable.releaseLock(2);
+    EXPECT_EQ(0UL, lockTable.buckets[0].entries[2]);
+}
+
+TEST_F(LockTableTest, releaseLock_overflow) {
+    lockTable.buckets[1].next->entries[3] = 3;
+
+    EXPECT_EQ(3UL, lockTable.buckets[1].next->entries[3]);
+    lockTable.releaseLock(3);
+    EXPECT_EQ(0UL, lockTable.buckets[1].next->entries[3]);
+}
+
+TEST_F(LockTableTest, releaseLock_notFound) {
+    lockTable.releaseLock(42);
+}
+
+TEST_F(LockTableTest, tryAcquireLock_basic) {
+    EXPECT_EQ(0UL, lockTable.buckets[1].entries[1]);
+    EXPECT_TRUE(lockTable.tryAcquireLock(1));
+    EXPECT_EQ(1UL, lockTable.buckets[1].entries[1]);
+    EXPECT_FALSE(lockTable.tryAcquireLock(1));
+}
+
+TEST_F(LockTableTest, tryAcquireLock_addCacheLine) {
+    EXPECT_TRUE(lockTable.buckets[0].next == NULL);
+    for (uint64_t i = 1; i <= LockTable::ENTRIES_PER_CACHE_LINE; i++) {
+        lockTable.acquireLock(2 * i);
+    }
+    EXPECT_FALSE(lockTable.buckets[0].next == NULL);
+}
```

Comment:
Do the tests above include a case where the first available entry is not the first entry in the CacheLine?